service/statistics/AnalyticsEvents.js

```diff
@@ -1,447 +1,426 @@
+ * constant. Otherwise use a factory function.
  *
- * TODO: document, reformat
+ * Note that the AnalyticsAdapter uses the events passed to its functions for
+ * its own purposes, and might modify them. Because of this factory functions
+ * should create new objects.
  *
- * Full event names (uncertain):
- * conference.muc.joined (???)
- * conference.sharingDesktop.start (???)
- * conference.sharingDesktop.stop (???)
- * xmpp.attached (???)
- * xmpp.attaching (???)
- * xmpp.connected (???)
- * xmpp.connecting (???)
- * xmpp.session-initiate (???)
  */
-export const _CONNECTION_TIMES_ = '';
 
 /**
- * TODO: document, reformat (group together with other ICE events)
- *
- * Known full event names:
- * ice.initiator.checksDuration
- * ice.responder.checksDuration
- * p2p.ice.initiator.checksDuration
- * p2p.ice.responder.checksDuration
- *
- * Properties: value
+ * The constant which identifies an event of type "operational".
+ * @type {string}
  */
-export const _ICE_CHECKING_DURATION = 'checksDuration';
+export const TYPE_OPERATIONAL = 'operational';
 
 /**
- * TODO: document, reformat
- *
- * Known full event names:
- * ice.checking
- * ice.closed
- * ice.completed
- * ice.connected
- * ice.disconnected
- * ice.failed
- * p2p.ice.checking
- * p2p.ice.closed
- * p2p.ice.completed
- * p2p.ice.connected
- * p2p.ice.disconnected
- * p2p.ice.failed
- *
- * Properties: value
+ * The constant which identifies an event of type "page".
+ * @type {string}
  */
-export const _ICE_CONNECTION_STATE_ = 'ice';
+export const TYPE_PAGE = 'page';
 
 /**
- * TODO: document, reformat (group together with other ICE events)
- *
- * Known full event names:
- * ice.initiator.establishmentDuration
- * ice.responder.establishmentDuration
- * p2p.ice.initiator.establishmentDuration
- * p2p.ice.responder.establishmentDuration
- *
- * Properties: value
+ * The constant which identifies an event of type "track".
+ * @type {string}
  */
-export const _ICE_ESTABLISHMENT_DURATION = 'establishmentDuration';
+export const TYPE_TRACK = 'track';
 
 /**
- * TODO: document, reformat
- *
- * Known full event names:
- * ice.initiator.gatheringDuration
- * ice.responder.gatheringDuration
- * p2p.ice.initiator.gatheringDuration
- * p2p.ice.responder.gatheringDuration
- *
- * Properties: value
+ * The constant which identifies an event of type "ui".
+ * @type {string}
  */
-export const _ICE_GATHERING_DURATION = 'gatheringDuration';
+export const TYPE_UI = 'ui';
 
 /**
- * TODO: document, reformat
- *
- * Known full event names:
- * audio.no_data_from_source
- * video.no_data_from_source
+ * Events with source 'sporadic' are sent from various places in the code, and
+ * indicate things which happen any action from the user. These events do not
+ * happen in every conference, and do not happen periodically. They are grouped
+ * together by this value for the 'source' attribute, because they may be of
+ * interest when looking for events in a particular conference.
```

Comment:
What kind of interest are you talking about that gets solved by using a sporadic group? Could you give me an example?

Reply:
When I started to write this I thought that I would come across more events for this group. But I didn't. I'll remove this.